lib/puppet/resource_api/glue.rb

```diff
@@ -41,6 +41,13 @@ def to_hierayaml
       YAML.dump('type' => { title => attributes }).split("\n").drop(2).join("\n") + "\n"
     end
 
+    def to_json
```

Comment:
Is there something we include or a base class we could use that would make this just work?

Reply:
I haven't found where `to_json` is originally defined, and the whole point of the `glue.rb` is to avoid the secret handshakes required to initialise puppet's `Resource` structs, and also to provide a living documentation of the _actual_ API required on those objects.